x/vesting/keeper/integration_test.go

```diff
@@ -538,25 +869,102 @@ func delegate(clawbackAccount *types.ClawbackVestingAccount, amount int64) error
+// and returns any error that the Eth vesting transaction decorator might return.
+func validateAnteForEthTxs(msgs ...sdk.Msg) error {
 	encodingConfig := encoding.MakeConfig(app.ModuleBasics)
 	txBuilder := encodingConfig.TxConfig.NewTxBuilder()
-	err = txBuilder.SetMsgs(msgEthereumTx)
+	err := txBuilder.SetMsgs(msgs...)
 	s.Require().NoError(err)
 
 	tx := txBuilder.GetTx()
 
 	// Call Ante decorator
-	dec := evmante.NewEthVestingTransactionDecorator(s.app.AccountKeeper)
+	dec := evmante.NewEthVestingTransactionDecorator(s.app.AccountKeeper, s.app.BankKeeper, s.app.EvmKeeper)
 	_, err = dec.AnteHandle(s.ctx, tx, false, nextFn)
 	return err
 }
+
+// deliverEthTxs is a helper function to deliver multiple messages with the same
+// private key.
+func deliverEthTxs(priv *ethsecp256k1.PrivKey, msgs ...sdk.Msg) error {
+	_, err := testutil.DeliverEthTx(s.ctx, s.app, priv, msgs...)
+	return err
+}
+
+// assertEthFails is a helper function that takes in 1 or more messages and checks
+// that they can neither be validated nor delivered.
+func assertEthFails(msgs ...sdk.Msg) {
+	insufficientUnlocked := "insufficient unlocked"
+
+	err := validateAnteForEthTxs(msgs...)
+	Expect(err).ToNot(BeNil())
+	Expect(strings.Contains(err.Error(), insufficientUnlocked))
+
+	// Sanity check that delivery fails as well
+	err = deliverEthTxs(nil, msgs...)
+	Expect(err).ToNot(BeNil())
+	Expect(strings.Contains(err.Error(), insufficientUnlocked))
+}
+
+// assertEthSucceeds is a helper function, that checks if 1 or more messages
+// can be validated and delivered.
+func assertEthSucceeds(testAccounts []TestClawbackAccount, funder sdk.AccAddress, dest sdk.AccAddress, amount math.Int, denom string, msgs ...sdk.Msg) {
+	numTestAccounts := len(testAccounts)
+
+	// Track starting balances for all accounts
+	granteeBalances := make(sdk.Coins, numTestAccounts)
+	funderBalance := s.app.BankKeeper.GetBalance(s.ctx, funder, denom)
+	destBalance := s.app.BankKeeper.GetBalance(s.ctx, dest, denom)
+
+	for i, grantee := range testAccounts {
+		granteeBalances[i] = s.app.BankKeeper.GetBalance(s.ctx, grantee.address, denom)
+	}
+
+	// Validate the AnteHandler passes without issue
+	err := validateAnteForEthTxs(msgs...)
+	Expect(err).To(BeNil())
+
+	// Expect delivery to succeed, then compare balances
+	err = deliverEthTxs(nil, msgs...)
+	Expect(err).To(BeNil())
+
+	fb := s.app.BankKeeper.GetBalance(s.ctx, funder, denom)
+	db := s.app.BankKeeper.GetBalance(s.ctx, dest, denom)
+
+	s.Require().Equal(funderBalance, fb)
+	s.Require().Equal(destBalance.AddAmount(amount).Amount.Mul(math.NewInt(int64(numTestAccounts))), db.Amount)
+
+	for i, account := range testAccounts {
+		gb := s.app.BankKeeper.GetBalance(s.ctx, account.address, denom)
+		// Use GreaterOrEqual because the gas fee is non-recoverable
+		s.Require().GreaterOrEqual(granteeBalances[i].SubAmount(amount).Amount.Uint64(), gb.Amount.Uint64())
+	}
+}
```

Comment:
let's also refactor these functions

Reply:
@fedekunze this is pretty much tailored to this specific case (array of clawback accounts passed in, check that the balance is increased, validating a specific ante handler, check for specific error string, ...)

I adjusted this to use the `testutil.DeliverEthTx` function but not too sure what else to refactor into the global testutils here?